parsl/tests/test_data/test_file_staging.py

```diff
@@ -1,88 +1,25 @@
-import os
 import pytest
+
 from parsl.app.app import bash_app
 from parsl.data_provider.files import File
 
 
 @bash_app
-def cat(inputs=[], outputs=[], stdout=None, stderr=None):
-    infiles = ' '.join([i.filepath for i in inputs])
-    return """echo {i}
-    cat {i} &> {o}
-    """.format(i=infiles, o=outputs[0])
-
-
-@pytest.mark.usefixtures('setup_data')
-@pytest.mark.issue363
-def test_files():
-
-    if os.path.exists('cat_out.txt'):
-        os.remove('cat_out.txt')
-
-    fs = [File('data/' + f) for f in os.listdir('data')]
-    x = cat(inputs=fs, outputs=[File('cat_out.txt')],
-            stdout='f_app.out', stderr='f_app.err')
-    d_x = x.outputs[0]
-    print(x.result())
-    print(d_x, type(d_x))
-
-
-@bash_app
-def increment(inputs=[], outputs=[], stdout=None, stderr=None):
-    # Place double braces to avoid python complaining about missing keys for {item = $1}
-    return """
-    x=$(cat {i})
-    echo $(($x+1)) > {o}
-    """.format(i=inputs[0], o=outputs[0])
+def cat(inputs=(), outputs=(), stdout=None, stderr=None):
+    infiles = " ".join(i.filepath for i in inputs)
+    return f"cat {infiles} &> {outputs[0]}\n"
 
 
 @pytest.mark.staging_required
-def test_regression_200():
+def test_regression_200(tmp_path):
     """Regression test for #200. Pickleablility of Files"""
+    opath = tmp_path / "test_output.txt"
+    fpath = tmp_path / "test.txt"
 
-    if os.path.exists('test_output.txt'):
-        os.remove('test_output.txt')
-
-    with open("test.txt", 'w') as f:
-        f.write("Hello World")
-
-    fu = cat(inputs=[File("test.txt")],
-             outputs=[File("test_output.txt")])
-    fu.result()
-
-    fi = fu.outputs[0].result()
-    print("type of fi: {}".format(type(fi)))
-    with open(str(fi), 'r') as f:
-        data = f.readlines()
-        assert "Hello World" in data, "Missed data"
-
-
-@pytest.mark.staging_required
-def test_increment(depth=5):
-    """Test simple pipeline A->B...->N
-    """
-    # Create the first file
-    open("test0.txt", 'w').write('0\n')
-
-    # Create the first entry in the dictionary holding the futures
-    prev = File("test0.txt")
-    futs = {}
-    for i in range(1, depth):
-        if os.path.exists('test{0}.txt'.format(i)):
-            os.remove('test{0}.txt'.format(i))
-        print("Launching {0} with {1}".format(i, prev))
-        fu = increment(inputs=[prev],  # Depend on the future from previous call
-                       # Name the file to be created here
-                       outputs=[File("test{0}.txt".format(i))],
-                       stdout="incr{0}.out".format(i),
-                       stderr="incr{0}.err".format(i))
-        [prev] = fu.outputs
-        futs[i] = prev
-        print(prev.filepath)
+    fpath.write_text("Hello World")
+    f = cat(inputs=[File(str(fpath))], outputs=[File(str(opath))])
```

Comment:
it would make sense for File() to accept a pathlike-object, rather than fpath being cast to a string, I think... apparently it isn't, though. (the original implementation predates the pathlike interface which was introduced in Python 3.6, although time has moved on enough that pathlike exists in all supported Python versions now...)

no need for a change here, this is a just rambling.

Reply:
I noted this in an issue https://github.com/Parsl/parsl/issues/2785

Reply:
Upon merge, address via [PR #2806 "Teach File to be PathLike aware"](https://github.com/Parsl/parsl/pull/2806).